src/main/java/org/jabref/logic/importer/fetcher/CitationRelationFetcher.java

```diff
@@ -0,0 +1,137 @@
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import javafx.beans.property.DoubleProperty;
+import javafx.beans.property.SimpleDoubleProperty;
+import kong.unirest.json.JSONArray;
+import kong.unirest.json.JSONException;
+
+import org.jabref.logic.importer.EntryBasedFetcher;
+import org.jabref.logic.importer.FetcherException;
+import org.jabref.model.entry.BibEntry;
+import org.jabref.model.entry.field.StandardField;
+import org.jabref.preferences.JabRefPreferences;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Class to fetch for an articles citation relations on opencitations.net's API
+ */
+public class CitationRelationFetcher implements EntryBasedFetcher {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(org.jabref.logic.importer.fetcher.CitationRelationFetcher.class);
+    private SearchType searchType = null;
+    private static final String BASIC_URL = "https://opencitations.net/index/api/v1/metadata/";
+    private final DoubleProperty progress = new SimpleDoubleProperty(0);
+
+    /**
+     * Possible search methods
+     */
+    public enum SearchType {
+        CITING("reference"),
+        CITEDBY("citation");
+
+        public final String label;
+
+        SearchType(String label) {
+            this.label = label;
+        }
+    }
+
+    public CitationRelationFetcher(SearchType searchType) {
+        this.searchType = searchType;
+        progress.set(0);
+    }
+
+    /**
+     * Executes the search method associated with searchType
+     *
+     * @param entry Entry to search relations for
+     * @return List of BibEntries found
+     */
+    @Override
+    public List<BibEntry> performSearch(BibEntry entry) throws FetcherException {
+        String doi = entry.getField(StandardField.DOI).orElse("");
+        if (searchType != null) {
+            List<BibEntry> list = new ArrayList<>();
+            try {
+                LOGGER.info("Search: " + BASIC_URL + doi);
+                JSONArray json = readJsonFromUrl(BASIC_URL + doi);
+                if (json == null) {
+                    throw new FetcherException("No internet connection! Please try again.");
+                } else if (json.isEmpty()) {
+                    return list;
+                }
+                LOGGER.info("API Answer: " + json.toString());
+                String[] items = json.getJSONObject(0).getString(searchType.label).split("; ");
+                if (!Arrays.equals(items, new String[]{""})) {
+                    int i = 1;
+                    for (String item : items) {
+                        LOGGER.info("Current Item " + i + "/" + items.length);
+                        setProgress(i, items.length);
+                        if (!doi.equals(item) && !item.equals("")) {
+                            DoiFetcher doiFetcher = new DoiFetcher(JabRefPreferences.getInstance().getImportFormatPreferences());
```

Comment:
Although this works, it results in quite a lot of requests to resolve the bibliographic information for each of the cited/citing entries. I fear we might get blocked because of this. Sadly the opencitation rest api doesn't allow to directly get the bibliographic infos (do I miss something?), but their SPARQL endpoint does allow to get them: https://opencitations.net/sparql. Can you please look into how much work it would be to use this?

Reply:
Of course a faster solution would be very good to improve the usability.
We will have a look into this, and come back to your concern 👍🏼 

Maybe one can use the DOI-Fetcher as a backup, when the SPARQL endpoint doesnt work?

Reply:
We updated the fetcher using OpenCitations API again to search for bibliographic data of all entries at once. The REST API's of OpenCitations.net return the same data as their SPARQL endpoints, so we thought it would be easier to use the REST API again.